Essentials/src/main/java/net/essentialsx/api/v2/events/HomeModifyEvent.java

```diff
@@ -0,0 +1,137 @@
+                create ? null : name, // oldName
+                create ? HomeModifyCause.CREATE : HomeModifyCause.DELETE);
+    }
+
+    public HomeModifyEvent(IUser user, IUser homeOwner, String oldName, String newName, Location location) {
+        this(user, homeOwner, location, location, newName, oldName, HomeModifyCause.RENAME);
+    }
+
+    public HomeModifyEvent(IUser user, IUser homeOwner, String name, Location oldLocation, Location newLocation) {
+        this(user, homeOwner, newLocation, oldLocation, name, name, HomeModifyCause.UPDATE);
+    }
+
+    public HomeModifyEvent(IUser user, IUser homeOwner, Location newLocation, Location oldLocation, String newName, String oldName, HomeModifyCause cause) {
+        this.user = user;
+        this.homeOwner = homeOwner;
+        this.newLocation = newLocation;
+        this.oldLocation = oldLocation;
+        this.newName = newName;
+        this.oldName = oldName;
+        this.cause = cause;
+    }
+
+    /**
+     * Gets the user who modified the home or null if the console modified the home.
+     * @return The user who modified the home or null.
+     */
+    public IUser getUser() {
+        return user;
+    }
+
+    /**
+     * Gets the owner of the home being modified.
+     * @return The user who owns the home.
+     */
+    public IUser getHomeOwner() {
+        return homeOwner;
+    }
+
+    /**
+     * Returns the location of the home when {@link #getCause()} returns {@link HomeModifyCause#CREATE} or {@link HomeModifyCause#RENAME},
+     * returns the updated location of the home if it returns {@link HomeModifyCause#UPDATE}, or returns null if it returns {@link HomeModifyCause#DELETE}.
+     * @return The location of the home or null.
+     */
+    public Location getNewLocation() {
+        return newLocation;
+    }
+
+    /**
+     * Returns the location of the home when {@link #getCause()} returns {@link HomeModifyCause#RENAME} or {@link HomeModifyCause#DELETE},
+     * returns the previous location of the home if it returns {@link HomeModifyCause#UPDATE}, or returns null if it returns {@link HomeModifyCause#CREATE}.
+     * @return The location of the home or null.
+     */
+    public Location getOldLocation() {
+        return oldLocation;
+    }
+
+    /**
+     * Returns the name of the home when {@link #getCause()} returns {@link HomeModifyCause#CREATE} or {@link HomeModifyCause#UPDATE},
+     * returns the updated name if it returns {@link HomeModifyCause#RENAME}, or returns null if it returns {@link HomeModifyCause#DELETE}.
+     * @return The name of the home or null.
+     */
+    public String getNewName() {
+        return newName;
+    }
+
+    /**
+     * Returns the name of the home when {@link #getCause()} returns {@link HomeModifyCause#UPDATE},
+     * returns the previous name if it returns {@link HomeModifyCause#RENAME}, or returns null if it returns {@link HomeModifyCause#DELETE} or {@link HomeModifyCause#CREATE}.
```

Comment:
If I understand the constructors right, this should return the name on delete? As described in the doc right now, you would never be able to get the name of a deleted home

Reply:
Doc is wrong, correct